src/AsyncMonitor.ts

```diff
@@ -56,7 +56,10 @@ function lock(...requests: Array<MultiLockRequest>) {
     } else if (descriptor.set != null) {
       kind = 'set';
     }
-    const f: Function = descriptor[kind]; // eslint-disable-line @typescript-eslint/ban-types
+
+    type AnyFn = (...args: unknown[]) => unknown;
```

Comment:
When defining an array type we always use `Array<T>` and not `T[]` by convention.